signals.go

```diff
@@ -28,12 +28,16 @@ func newSignalHandler(enableSubreaper bool, notifySocket *notifySocket) *signalH
 	// ensure that we have a large buffer size so that we do not miss any signals
 	// in case we are not processing them fast enough.
 	s := make(chan os.Signal, signalBufferSize)
-	// handle all signals for the process.
-	signal.Notify(s)
-	return &signalHandler{
-		signals:      s,
-		notifySocket: notifySocket,
-	}
+	handler := make(chan *signalHandler)
+	go (func() {
```

Comment:
nit: the `(` before the `func` is not needed.

Reply:
OMG, this fully exposes that I wrote quite a few typescripts today.